### PR TITLE
Changed knuth_morris_pratt to be consistent with str.find()

### LINKED ISSUE #9077: Returning index instead of boolean when knuth_morris_pratt matches.
In [knuth morris pratt algorithm](https://github.com/TheAlgorithms/Python/blob/master/strings/knuth_morris_pratt.py), I think it's better to return the "index" of the matched substring in case of success, and `-1` in case of failure, so that it would be consistent with the `str.find()` in Python. `True` and `False` aren't that helpful.

### PR DESCRIPTION
**Pull request by Akashram28**
### Describe your change:

function kmp() now returns the first occurrence of the pattern. If the pattern does not exist in the given text, it returns -1 

See https://github.com/TheAlgorithms/Python/issues/9077

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #9077".

### CODE REVIEW

**Comment by rohan472000** [comment]
Change bool return type to int.

**Comment by rohan472000** [comment]
In your PR description, you should tick whatever suits your PR but don't delete other checkboxes, also tick that checkbox for bug fixing or doc change which u deleted.

File: strings/knuth_morris_pratt.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from __future__ import annotations
 
 
-def kmp(pattern: str, text: str) -> bool:
+def kmp(pattern: str, text: str) -> int:
```
Comment (rohan472000) [inline strings/knuth_morris_pratt.py:4]:
```suggestion
def knuth_morris_pratt(pattern: str, text: str) -> int:
```

Comment (rohan472000) [reply strings/knuth_morris_pratt.py:4]: As `kmp` is vague not understandable to others.

**Review by rohan472000 — CHANGES_REQUESTED** [review]

**Comment by rohan472000** [comment]
@cclauss , kindly review it.

**Comment by amirsoroush** [comment]
Thanks. "doctests" are more desirable than print statements. While we're at it, let's write some tests instead.

**Comment by amirsoroush** [comment]
Just to note, you can use some keywords in your commit message to link a pull request to an issue, that way if your PR is accepted, the linked issue gets closed automatically.

**Review by rohan472000 — APPROVED** [review]
LGTM

**Comment by cclauss** [comment]
I added doctests that fail...
```
=================================== FAILURES ===================================
___________ [doctest] strings.knuth_morris_pratt.knuth_morris_pratt ____________
009     1) Preprocess pattern to identify any suffixes that are identical to prefixes
010 
011         This tells us where to continue from if we get a mismatch between a character
012         in our pattern and the text.
013 
014     2) Step through the text one character at a time and compare it to a character in
015         the pattern updating our location within the pattern if necessary
016 
017     >>> kmp = "knuth_morris_pratt"
018     >>> knuth_morris_pratt(kmp, "kn") == kmp.find("kn")
Expected:
    True
Got:
    False
```

**Comment by rohan472000** [comment]
> I added doctests that fail...
> 
> ```
> =================================== FAILURES ===================================
> ___________ [doctest] strings.knuth_morris_pratt.knuth_morris_pratt ____________
> 009     1) Preprocess pattern to identify any suffixes that are identical to prefixes
> 010 
> 011         This tells us where to continue from if we get a mismatch between a character
> 012         in our pattern and the text.
> 013 
> 014     2) Step through the text one character at a time and compare it to a character in
> 015         the pattern updating our location within the pattern if necessary
> 016 
> 017     >>> kmp = "knuth_morris_pratt"
> 018     >>> knuth_morris_pratt(kmp, "kn") == kmp.find("kn")
> Expected:
>     True
> Got:
>     False
> ```
> 
> As discussed in CONTRIBUTING.md, you can run these on your own computer with: `python3 -m doctest -v strings/knuth_morris_pratt.py`

I'm getting false too for that doctest.

**Comment by rohan472000** [comment]
image
individually both are giving false, but why not true when doing `==`.

**Comment by rohan472000** [comment]
while testing the doctests added by @cclauss with existing code of `knuth_morris_pratt.py` in repo, it is also failing.

**Comment by Akashram28** [comment]
> image in
knuth_morris_pratt
dividually both are giving false, but why not true when doing `==`.

The pattern is the first parameter, the text is the second

**Comment by cclauss** [comment]
If we are going to be consistant with [`str.find()`](https://docs.python.org/3/library/stdtypes.html#str.find) then we should change `knuth_morris_pratt(sub, text)` to `knuth_morris_pratt(text, sub)` to match the calling conventions and `knuth_morris_pratt(text, sub) == text.find(sub)` should always be True.

**Pull request closed without merging**